tests/Doctrine/Migrations/Tests/Tools/Console/Command/DoctrineCommandTest.php

```diff
@@ -44,6 +44,39 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         );
     }
 
+    public function testCommandNotThrowingFrozenException()
+    {
+        $configurationLoader = $this->createMock(ConfigurationLoader::class);
+        $configurationLoader->method('getConfiguration')->willReturn(new Configuration());
+
+        $dependencyFactory = DependencyFactory::fromConnection(
+            $configurationLoader,
+            $this->createMock(ConnectionLoader::class)
+        );
+
+        $command = new class ($dependencyFactory) extends DoctrineCommand {
+            protected function execute(InputInterface $input, OutputInterface $output): int
+            {
+                return 0;
+            }
+        };
+        $commandTester = new CommandTester($command);
+
+        // execute once, this will freeze the dependencies.
+        $commandTester->execute(
```

Comment:
im not sure if this test makes sense... is it possible to execute two times the same command ? ( i guess that you are using it as a service, not CLI, right?)

Reply:
Yes, so basically services are instantiated, and second ran of command uses same services, and one of them (DependencyFactory) is frozen and throws exception on second ran.

Reply:
Basically like that: https://github.com/Warxcell/files/blob/5.0/tests/Functional/AbstractFunctionalTest.php#L68-L81